test/performance/core/simd/simd_iota_benchmark.cpp

```diff
@@ -0,0 +1,185 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <seqan3/std/ranges>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/core/simd/simd_algorithm.hpp>
+#include <seqan3/core/simd/simd.hpp>
+#include <seqan3/core/simd/view_iota_simd.hpp>
+
+// ============================================================================
+//  simd_iota_iterator
+// ============================================================================
+
+template <typename simd_t>
+struct simd_iota_iterator_function
+{
+    using scalar_type = typename seqan3::simd::simd_traits<simd_t>::scalar_type;
+
+    seqan3::views::iota_simd<simd_t> simd_iota_view;
+
+    simd_iota_iterator_function(size_t const end_index) :
+        simd_iota_view{static_cast<scalar_type>(0), static_cast<scalar_type>(end_index)}
+    {}
+
+    void operator()(simd_t & count) const
+    {
+        for (auto simd_index : simd_iota_view)
+            count += simd_index;
+    }
+};
+
+// ============================================================================
+//  iota_transform
+// ============================================================================
+
+template <typename simd_t>
+struct transform_to_simd
+{
+    template <typename index_t>
+    simd_t operator()(index_t const index) const
+    {
+        return seqan3::simd::fill<simd_t>(index);
+    }
+};
+
+template <typename simd_t>
+struct iota_transform_function
+{
+    using iota_transform_view_type = decltype(std::views::iota(0u, static_cast<size_t>(1))
+                                            | std::views::transform(transform_to_simd<simd_t>{}));
```

Comment:
```suggestion
                                            | std::views::transform(seqan3::simd::fill<simd_t>));
```

I guess this does not work, right?